Unit-01/03-templating/app.py

```diff
@@ -0,0 +1,34 @@
+from flask import Flask, render_template, request
+
+app = Flask(__name__)
+
+#PART 1
+@app.route('/person/<name>/<int:age>')
+def person(name, age):
+	return render_template('person.html', name=name, age=age)
+
+    
+# PART2
+@app.route('/calculate')
+def calculate():
+	return render_template('calc.html')
+
+@app.route('/math', methods=['POST'])
+def math():
+	num1 = request.form.get('num1')
+	num2 = request.form.get('num2')
+	if request.form.get('calculation') == 'add':
+		value = int(num1) + int(num2)
+		return str(value)
+	elif request.form.get('calculation') == 'subtract':
```

Comment:
lots of code duplication, consider doing some abstraction like this:

```python
def add(a, b):
    return a + b
def subtract(a, b):
    return a - b
def multiply(a, b):
    return a * b
def divide(a / b):
    return a / b

operations = {
    "add": add,
    "subtract": subtract,
    "multiply": multiply,
    "divide": divide
}

# @route handler
def math(operation, num1, num2):
    if operation in operations:
        return f"{operations[operation](num1, num2)}"
```

This is a more declarative style, where you can declare your functions and data structures at the time and simplify the business logic part (i.e. the routing, in this case)